music_assistant/providers/phishin/provider.py

```diff
@@ -0,0 +1,1061 @@
+                            total_pages = shows_data.get("total_pages", 1)
+
+                            if current_page >= total_pages or len(venue_shows) >= limit:
+                                break
+
+                            page += 1
+
+                    except ProviderUnavailableError:
+                        self.logger.warning(
+                            "Failed to fetch shows for venue %s: API unavailable", venue_slug
+                        )
+
+                if venue_shows:
+                    search_data["venue_shows"] = venue_shows
+
+            artists, albums, tracks, playlists = parse_search_results(
+                self, search_data, media_types, search_query
+            )
+
+            return SearchResults(
+                artists=artists[:limit] if MediaType.ARTIST in media_types else [],
+                albums=albums[:limit] if MediaType.ALBUM in media_types else [],
+                tracks=tracks[:limit] if MediaType.TRACK in media_types else [],
+                playlists=playlists[:limit] if MediaType.PLAYLIST in media_types else [],
+            )
+        except (MediaNotFoundError, ProviderUnavailableError):
+            raise
+        except Exception as err:
+            self.logger.error("Search failed for query '%s': %s", search_query, err)
+            raise ProviderUnavailableError(f"Search error: {err}") from err
+
+    async def get_library_artists(self) -> AsyncGenerator[Artist, None]:
+        """Retrieve library artists from the provider."""
+        yield await get_phish_artist(self)
+
+    async def get_library_albums(self) -> AsyncGenerator[Album, None]:
+        """Retrieve library albums (shows) from the provider."""
+        page = 1
+        per_page = 50
+        max_pages = 20  # 1000 albums max for UI performance
+
+        try:
+            while page <= max_pages:
+                shows_data = await api_request(
+                    self,
+                    ENDPOINTS["shows"],
+                    params={
+                        "page": page,
+                        "per_page": per_page,
+                        "audio_status": "complete_or_partial",
+                    },
+                )
+
+                shows = shows_data.get("shows", [])
+                if not shows:
+                    break
+
+                for show in shows:
+                    if show.get("audio_status") in ["complete", "partial"]:
+                        yield show_to_album(self, show)
+
+                if len(shows) < per_page:
+                    break
+
+                page += 1
+
+        except (MediaNotFoundError, ProviderUnavailableError):
+            raise
+        except Exception as err:
+            self.logger.error("Failed to get library albums: %s", err)
+            raise ProviderUnavailableError(f"Library albums error: {err}") from err
+
+    async def get_library_tracks(self) -> AsyncGenerator[Track, None]:
```

Comment:
So, what I'm wondering is if we should return all these items as library items ?
The definition of that is that the user added them to the library/favorites of the streaming provider but that is not the case here. You are just returning the whole catalog as library which I think is maybe not the right way to do this ?

Should we perhaps treat this the same as Radio Browser where you can simply Browse all available items and then manually add something you like to your MA library ?